Bump milli to v0.37.5

curquiza opened the pull request:
Fixes #3021

loiclec commented:
The dependencies are correct, but do you know why there is a clippy error just now?

curquiza commented:
Clippy error is due to a rust upgrade, we fixed it on `main` (https://github.com/meilisearch/meilisearch/pull/3262), and now I fixed it here: https://github.com/meilisearch/meilisearch/pull/3268

curquiza commented:
bors merge
